packages/sdk-core/src/modules/converter/StackTraceConverter.ts

```diff
@@ -0,0 +1,161 @@
+import { BacktraceStackFrame } from '../../model/data/BacktraceStackTrace';
+import { JavaScriptEngine } from '../../model/data/JavaScriptEngine';
+
+export class StackTraceConverter {
+    public readonly UNKNOWN_FRAME = 'unknown';
+    private readonly ANONYMOUS_FRAME: Record<JavaScriptEngine, string> = {
+        ['firefox']: 'debugger eval code',
+        ['v8']: '<anonymous>',
+        ['safari']: '[native code]',
+    };
+    constructor(private readonly _lang: JavaScriptEngine) {}
+
+    public convert(message: string, stackTrace: string): BacktraceStackFrame[] {
+        const result: BacktraceStackFrame[] = [];
+        if (!stackTrace) {
+            return result;
+        }
+        let stackFrames = stackTrace.split('\n');
+        const errorHeader = message.split('\n');
+
+        // remove error header from stack trace - if the error header exists
+        if (stackFrames[0].indexOf(errorHeader[0]) !== -1) {
+            stackFrames = stackFrames.slice(errorHeader.length);
+        }
+
+        for (const stackFrame of stackFrames) {
+            const normalizedStackFrame = stackFrame.trim();
+            const frame = this.parseFrame(normalizedStackFrame);
+            if (this.ANONYMOUS_FRAME[this._lang] === frame.library) {
+                frame.library = this.UNKNOWN_FRAME;
+            }
+            result.push(frame);
+        }
+
+        return result;
+    }
+
+    private parseFrame(stackFrame: string): BacktraceStackFrame {
+        switch (this._lang) {
+            case 'firefox': {
+                return this.parseFirefoxStackFrame(stackFrame);
+            }
+            case 'safari': {
+                return this.parseSafariStackFrame(stackFrame);
+            }
+            default: {
+                return this.parseV8StackFrame(stackFrame);
+            }
+        }
+    }
+
+    private parseSafariStackFrame(stackFrame: string): BacktraceStackFrame {
+        const functionSeparator = '@';
+        const functionSeparatorIndex = stackFrame.indexOf(functionSeparator);
+        if (functionSeparatorIndex === -1) {
+            return {
+                funcName: stackFrame,
+                library: this.UNKNOWN_FRAME,
+            };
+        }
+
+        const functionName = stackFrame.substring(0, functionSeparatorIndex) ?? this.UNKNOWN_FRAME;
```

Comment:
From what I've seen in Safari, if there's nothing before `@`, it means that the function was anonymous.

Reply:
done.